guides/adr/0012-formalize-accredited-provider-relationships.md

```diff
@@ -0,0 +1,380 @@
+
+
+| Provider | Is accredited? | Is accredited by |
+| ---      | ---            | ---              |
+| A        | Yes            | -                |
+| B        | Yes            | A                |
+| C        | No             | B                |
+
+
+| Provider | Is accredited? | "Is accredited by" | asdf | Who ratifies their courses? |
+| ---      | ---            | ---                | ---  | ---                         |
+| C76      | Yes            | -                  |      | No courses                  |
+| 2DB      | Yes            | C76                |      | C76 (all)                   |
+| 1AV      | No             | 2DB                |      | 2DB (all)                   |
+
+
+* In Publish, a provider "is accredited by" is a relationship that exists solely to filter the accredited providers a training provider can choose from when selecting an accredited provider for their course.
+
+This relationship does not exist outside of Publish in concrete terms. In Apply:
+
+Provider A is the accredited provider of Provider B only if Provider B is running courses that are ratified by Provider A.
+
+Example: 
+https://github.com/DFE-Digital/publish-teacher-training/pull/4589/files
+https://github.com/DFE-Digital/publish-teacher-training/blob/b7fcf16c39e087706d2b076c3515c7a946126a45/db/data/20241011081455_update_accredited_providers.rb
+Question:
+- Which is the ratifying provider of the courses run by Provider B and Provider C?
+
+
+This feels like an abuse of the accredited provider association, or a mislabelling of the relationship.
+
+### Unconventional entity associations
+
+We have a database table `provider` that has an implicit self-join relationship based on another column `accrediting_provider`.
+The association exists through a column called `accredited_provider_enrichments`.
+
+The relationship that currently exists is unconventional with respect to a normalised relational database.
+
+This is explained more in [Option 1](#1.-leave-the-relationships-as-they-are).
+
+
+### Organisations
+
+Publish used to employ a system of 'Organisations'. A number of Providers would be grouped by all being a member of an organisation. 
+Users would also be members of the organisation. This architecture has been abandoned but there are still remnants of this system in the codebase.
+
+
+### Definitions
+
+ - A Training Provider
+     * A provider that runs training courses directly.
+     * |or|
+     * A provider that has an subordinate assocation to another provider (accredited).
+ - An Accredited Provider
+     * A provider that has an authoritative association to another provider (training).
+ - Accrediting Provider
+     * A provider that accredits one or more courses on behalf of one or more training provider.
+ - An Accredited Provider Relationship
+     * A directional association between an accredited provider and another provider.
+ - Lead partner
```

Comment:
Lead partner is a term used when accredited providers lost their accreditation. 

We previously had 'Lead schools,' which were logical groupings of schools with one designated as the administrator, hence 'lead school'. 

Instead of creating a new category for the accredited providers that lost accreditation, they were merged with the school grouping and called 'lead partners'.

In the UI, we use 'training partner' not least because they partner with accredited providers to deliver training. i.e., it's a more meaningful term.

Reply:
I just mean to say we haven't updated the terminology in the codebase from Lead school to Lead partner